docs/cluster_management.md

```diff
@@ -2,45 +2,76 @@
+More can be seen by running particulars app help.
 
-### Switching your cluster's storage API version
+If you want to change the value of particular flag, e.g. `max-pod`, you need to find a file used by Salt to set Kubelet flags. This particular file is `cluster/saltbase/salt/kubelet/default`. In this file you can modify the line with DAEMON_ARGS by adding the value for the flag you want, e.g. change the DAEMON_ARGS line to
+```
+DAEMON_ARGS="{{daemon_args}} {{api_servers_with_port}} {{debugging_handlers}} {{hostname_override}} {{cloud_provider}} {{config}} --allow_privileged={{pillar['allow_privileged']}} {{pillar['log_level']}} {{cluster_dns}} {{cluster_domain}} {{docker_root}} {{kubelet_root}} {{configure_cbr0}} {{cgroup_root}} {{system_container}} {{pod_cidr}} --max-pods=42"
+```
 
-KUBE_API_VERSIONS env var controls the API versions that are supported in the cluster. The first version in the list is used as the cluster's storage version. Hence, to set a specific version as the storage version, bring it to the front of list of versions in the value of KUBE_API_VERSIONS.
+If you want to have more flexibility for configuring your cluster, you'll need to learn more about the [Salt stack](http://saltstack.com/).
 
-### Switching your config files to a new API version
+## Upgrading a cluster
 
-You can use the kube-version-change utility to convert config files between different API versions.
+GKE handles cluster upgrades for you, so you don't need to worry about. If you already a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and most of other providers.
 
+If your cluster is running on GCE and you have a repository with the version of Kubernetes you want to push to your cluster, you just need to do the following:
+```
+make release
+cluster/kube-push.sh
 ```
-$ hack/build-go.sh cmd/kube-version-change
-$ _output/local/go/bin/kube-version-change -i myPod.v1beta3.yaml -o myPod.v1.yaml
+or equivalent
 ```
+hack/dev-build-and-push.sh
+```
+This will build new version of Kubernetes, run basic unit and integration tests on it, and if all tests pass upgrade running Kubernetes to the version you just built.
+
+## Scaling the running cluster
+
+When the Kubelet is given masters address it'll self register in running Kubernetes cluster.
+
+If your cluster runs out short on resources you can easily add more machines to it. If you're using GCE or GKE it is done by resizing Instance Group managing your Nodes. It's done by modifying number of instances on `Compute > Compute Engine > Instance groups > your group > Edit group` page. Instance Group will take care of putting appropriate image on new machines and start them, while Kubelet will register its Node with API server to make it available for scheduling.
```

Comment:
You should link to cloud-console or something here - it's no clear what "page" you mean.